pkg/pfconfig/basicstationlns/basicstationlns.go

```diff
@@ -0,0 +1,181 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package basicstationlns implements the JSON configuration for the Basic Station `router_config` message.
+package basicstationlns
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"go.thethings.network/lorawan-stack/pkg/band"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"go.thethings.network/lorawan-stack/pkg/frequencyplans"
+	"go.thethings.network/lorawan-stack/pkg/pfconfig/shared"
+)
+
+const (
+	configHardwareSpecPrefix = "sx1301"
+)
+
+var errFrequencyPlan = errors.DefineInvalidArgument("frequency_plan", "invalid frequency plan `{name}`")
+
+// DataRates encodes the available datarates of the channel plan for the Station in the format below:
+// [0] -> SF (Spreading Factor; Range: 7...12 for LoRa, 0 for FSK)
+// [1] -> BW (Bandwidth; 125/250/500 for LoRa, ignored for FSK)
+// [2] -> DNONLY (Downlink Only; 1 = true, 0 = false)
+type DataRates [16][3]int
+
+// RouterConfig contains the router configuration.
+// This message is sent by the Gateway Server.
+type RouterConfig struct {
+	NetID          []int                 `json:"NetID"`
+	JoinEUI        [][]int               `json:"JoinEui"`
+	Region         string                `json:"region"`
+	HardwareSpec   string                `json:"hwspec"`
+	FrequencyRange []int                 `json:"freq_range"`
+	DataRates      DataRates             `json:"DRs"`
+	SX1301Config   []shared.SX1301Config `json:"sx1301_conf"`
+
+	// These are debug options to be unset in production gateways.
+	NoCCA       bool `json:"nocca"`
+	NoDutyCycle bool `json:"nodc"`
+	NoDwellTime bool `json:"nodwell"`
+
+	MuxTime float64 `json:"MuxTime"`
+}
+
+// MarshalJSON implements json.Marshaler.
+func (cfg RouterConfig) MarshalJSON() ([]byte, error) {
+	type Alias RouterConfig
+	return json.Marshal(struct {
+		Type string `json:"msgtype"`
+		Alias
+	}{
+		Type:  "router_config",
+		Alias: Alias(cfg),
+	})
+}
+
+// GetRouterConfig returns the routerconfig message to be sent to the gateway.
+// Currently as per the basic station docs, all frequency plans have to be from the same region (band) https://doc.sm.tc/station/tcproto.html#router-config-message.
+func GetRouterConfig(fps []*frequencyplans.FrequencyPlan, isProd bool, dlTime time.Time) (RouterConfig, error) {
+	for _, fp := range fps {
+		if err := fp.Validate(); err != nil {
+			return RouterConfig{}, errFrequencyPlan
+		}
+	}
+	cfg := RouterConfig{}
```

Comment:
```suggestion
	conf := RouterConfig{}
```
https://github.com/TheThingsNetwork/lorawan-stack/blob/master/DEVELOPMENT.md#single-word-entities